scripts/downloads.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**This file checks what user is logged in and if the
+ * user is admin. Then it checks if the file they want exits
+ * for the specific user. (For admins it doesnt check if the file exits since
+ * admins can view everyone's file) It checks what type of file
+ * it is and sends the appropriate headers to download the file.
+ * @author Oleg Rovinskiy
+ * @version 1.0
+ */
+session_start();
+require_once("../model/config.php");
+include '../model/Database.php';
+
+//sets the variables that were sent
+if($_SERVER['REQUEST_METHOD'] == 'POST'){
+    $_SESSION['item'] = $_POST['item'];
+    $_SESSION['file'] = $_POST['fileName'];
+    $_SESSION['isOgFile'] = $_POST['ogFile'];
+}//creates the file from the previous variables that were sent
+else {
+    $db = new Database();
+    $ext = strtolower(substr($_SESSION['file'], strpos($_SESSION['file'], '.')));
+    $type = '';
+    $checkFile = '';
+
+    //Checks for what type of file
+    switch ($ext) {
+        case 'pdf':
+            $type = 'application/pdf';
+            break;
+        case 'docx' || 'doc':
+            $type = 'application/vnd.openxmlformats-officedocument.wordprocessingml.document';
+            break;
+        case 'zip':
+            $type = 'application/zip';
+            break;
+        case 'jpeg':
+            $type = 'image/jpeg';
+            break;
+        case 'jpg':
+            $type = 'image/jpg';
+            break;
+        case 'png':
+            $type = 'image/png';
+            break;
+    }
+
+    //checks if file exits for the user
+    if($_SESSION['isAdmin'] !== 1) {
```

Comment:
Does this also prevent tutors form getting access to other tutor's forms? I can't tell if it actually prevents this.

Reply:
yes it does. this works because each time you logout all the sessions get destroyed. So if isAdmin is not set it checks the database for the file for the specific tutor that asked for it. If you check the checklist function you will see $_Session[tutoryear} = $tutoryear. This tells me which user is logged in